pkg/operator/status/status_controller.go

```diff
@@ -192,71 +188,20 @@ func OperatorConditionToClusterOperatorCondition(condition operatorv1.OperatorCo
 	}
 }
 
-func (c *StatusSyncer) Run(ctx context.Context, workers int) {
-	defer utilruntime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	klog.Infof("Starting StatusSyncer-" + c.clusterOperatorName)
-	defer klog.Infof("Shutting down StatusSyncer-" + c.clusterOperatorName)
-	if !cache.WaitForCacheSync(ctx.Done(), c.cachesToSync...) {
-		return
-	}
-
-	// start watching for version changes
-	go c.watchVersionGetter(ctx.Done())
-
-	// doesn't matter what workers say, only start one.
-	go wait.UntilWithContext(ctx, c.runWorker, time.Second)
-
-	<-ctx.Done()
-}
-
-func (c *StatusSyncer) watchVersionGetter(stopCh <-chan struct{}) {
+func (c *StatusSyncer) watchVersionGetterPostRunHook(ctx context.Context, syncCtx factory.SyncContext) error {
 	defer utilruntime.HandleCrash()
+	workQueueKey := factory.QueueKey("StatusSyncer_" + c.clusterOperatorName)
```

Comment:
again, don't have more than one place where the constant key is defined

Reply:
fixed.